### ISSUE TITLE
stream aggregation: support sharding

### ISSUE THREAD

**Issue by hagen1778** [VictoriaMetrics/VictoriaMetrics#4637]
### Is your feature request related to a problem? Please describe

Stream aggregation can be used in heavy loaded environments and may require extra resource for processing all the configured rules. In some cases, it is just impossible for one vmagent to handle all the load on a single instance. 

Can vmagents be sharded? One of the important requirements for aggregation is that one specific time series should be always aggregated by one specific vmagent. This isolation can be achieved for scraping via https://docs.victoriametrics.com/vmagent.html#scraping-big-number-of-targets. But it doesn't cover the case when user pushes data to vmagent, or consumes metrics from Kafka or other integrations.

### Describe the solution you'd like

> One of the important requirements for aggregation is that one specific time series should be always aggregated by one specific vmagent.

Would be great, if vmagent could "shard" received time series among configured remote-writes consistently. With this option, users could build a multi-layer topology where Level1 agents are responsible for consistent sharding, and Level2 agents - for aggregation:
![image](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/2902918/7c6cd879-9211-4df2-a5dd-9dc55bb543d7)

The sharding mode for vmagents could be enabled in the following way:
```
/path/to/vmagent -shardingMode -remoteWrite.url=https://dst-1/api/v1/write -remoteWrite.url=https://dst-2/api/v1/write
```

`-shardingMode` is a bool flag, disabled by default. Flag name is a subject for discussion.
When `-shardingMode` enabled, vmagent should apply a hash function for each received time series (pushed or pulled) to chose a corresponding `remoteWrite.url` for sending.  vmagent can do an extra check to verify that at least 2 `remoteWrite.url` destinations are configured when `-shardingMode` is set.

**Comment by valyala** [comment]
`vmagent` is able to shard outgoing series among the configured `-remoteWrite.url` remote storage systems after the commit 6b6b61137ffc5989a6c49adf1be61b46891bd764 . This commit adds support for `-remoteWrite.shardByURL` command-line flag. If this flag is set, then `vmagent` spreads evenly (e.g. shards) outgoing time series among remote storage systems specified via multiple `-remoteWrite.url` command-line options. This allows routing samples for the same time series to the same `-remoteWrite.url`, which, in turn, allows building horizontally scalable [stream aggregation](https://docs.victoriametrics.com/stream-aggregation.html) for [counter](https://docs.victoriametrics.com/keyConcepts.html#counter) and [histogram](https://docs.victoriametrics.com/keyConcepts.html#histogram) metrics. See [these docs](https://docs.victoriametrics.com/vmagent.html#sharding-among-remote-storages) for details.

The commit will be included in the next release. In the mean time it is possible to test the `-remoteWrite.shardByURL` functionality by building `vmagent` from this commit according to [these instructions](https://docs.victoriametrics.com/vmagent.html#how-to-build-from-sources) and verifying that the functionality works as expected.

**Comment by valyala** [comment]
The `-remoteWrite.shardByURL` command-line flag is available in `vmagent` starting from [v1.92.0](https://docs.victoriametrics.com/CHANGELOG.html). Closing this feature request as done.
